zaza/utilities/juju.py

```diff
@@ -279,11 +280,28 @@ def get_machine_series(machine, model_name=None):
     :returns: Juju series
     :rtype: string
     """
-    return get_machine_status(
+    status = get_machine_status(
         machine=machine,
-        key='series',
         model_name=model_name
     )
+    try:
+        if 'series' in status:
+            return status.get('series')
+    except KeyError:
```

Comment:
Nothing here seems to raise a `KeyError`. Since `get_machine_status` returns a plain dict without the `series` parameter, this could be simplified to:

```
if 'series' in status:
  return status['series']
```

Reply:
I thought so too and have tried, but libjuju unfortunately constructs an object that behaves subtly different than a `Dict`. The comment below is a testament :)